### PR TITLE
TN-2781 Integrate SDC service

### PR DESCRIPTION
**Pull request by ivan-c**
TN-2781
* Integrate SDC at service level

### CODE REVIEW

File: portal/tasks.py [hunk 1]
```diff
@@ -356,3 +356,32 @@ def celery_beat_health_check(**kwargs):
...
+    #from .trigger_states.empro_states import enter_user_trigger_critical_section
+    qnr = QuestionnaireResponse.query.get(questionnaire_response_id)
+
+    # TODO enable after merging
+    #enter_user_trigger_critical_section(user_id=qnr.subject_id)
+
+    qnr_json = qnr.as_sdc_fhir()
+
+    SDC_BASE_URL = current_app.config['SDC_BASE_URL']
+    response = post(f"{SDC_BASE_URL}/$extract", json=qnr_json)
```
Comment (pbugni) [inline portal/tasks.py:379]: this looks like a blocking call?  worth a comment?

**Review by pbugni — APPROVED** [review]
awesome progress

File: portal/views/assessment_engine.py [hunk 2]
```diff
@@ -1575,6 +1575,10 @@ def assessment_add(patient_id):
     db.session.commit()
     questionnaire_response.assign_qb_relationship(
         acting_user_id=current_user().id)
+
+    from ..tasks import extract_observations_task
```
Comment (pbugni) [inline portal/views/assessment_engine.py:1579]: shouldn't we limit this expensive call to QNRs that include the `sdc-questionnaire-observationExtract` extension?

File: portal/views/assessment_engine.py [hunk 3]
```diff
@@ -1575,6 +1575,10 @@ def assessment_add(patient_id):
...
+    extract_observations_task.apply_async(questionnaire_response_id=questionnaire_response.id)
```
Comment (pbugni) [inline portal/views/assessment_engine.py:1580]: don't we need to do this in the PUT / update QNR call as well?

Comment (ivan-c) [reply portal/views/assessment_engine.py]: Done! Thanks for the reminder

**Review by pbugni — APPROVED** [review]

File: portal/views/assessment_engine.py (hunk 2 above)
Comment (ivan-c) [reply portal/views/assessment_engine.py]: Thanks, as discussed added a hack and a TODO so we don't forget :crossed_fingers:

**Pull request merged**